### PR TITLE
fix(glad): use slug update only to golang packages

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description 
We have [logic](https://github.com/aquasecurity/vuln-list-update/blob/67f99a96ff13172d6d67c663b71d00fed3ff711c/glad/glad.go#L122-L128) to flatten nested packages.
But it works only for `golang`.
Other languages can miss some packages.

For example:
GitLab database has [json-smart](https://gitlab.com/gitlab-org/security-products/gemnasium-db/-/blob/ecd84f42eead545dc34b936bd1a24688fa86961e/maven/net.minidev/json-smart/CVE-2021-27568.yml) and [json-smart-mini](https://gitlab.com/gitlab-org/security-products/gemnasium-db/-/blob/ecd84f42eead545dc34b936bd1a24688fa86961e/maven/net.minidev/json-smart-mini/CVE-2021-27568.yml) packages.

Currently `vuln-list-update` writes `json-smart`, overwrite it data from `json-smart-mini` and doesn't save `json-smart-mini` package([vuln-list folder](https://github.com/aquasecurity/vuln-list/tree/3bbbf80ebe5c5c236cbeb4e0ae0f48cc5a47c902/glad/maven/net.minidev)).

<details><summary> list of overwritten maven files</summary>

``` bash
> ./org.apache.logging.log4j/log4j-core/CVE-2021-44228.json
> ./org.apache.logging.log4j/log4j-core/CVE-2021-44832.json
> ./org.apache.logging.log4j/log4j-core/CVE-2021-45046.json
> ./org.apache.logging.log4j/log4j-core/CVE-2021-4104.json
> ./org.apache.logging.log4j/log4j-core/CVE-2021-45105.json
> ./org.apache.logging.log4j/log4j-core/CVE-2020-9488.json
> ./org.apache.logging.log4j/log4j-core/CVE-2017-5645.json
22a30,33
> ./org.apache.logging.log4j/log4j-api/CVE-2021-44228.json
> ./org.apache.logging.log4j/log4j-api/CVE-2021-44832.json
> ./org.apache.logging.log4j/log4j-api/CVE-2021-45046.json
> ./org.apache.logging.log4j/log4j-api/CVE-2021-45105.json
57a69
> ./org.jbpm/jbpm-designer-client/CVE-2016-8608.json
121a134
> ./org.asynchttpclient/async-http-client-project/CVE-2017-14063.json
163a177,182
> ./org.apache.syncope/syncope-core/CVE-2020-1959.json
> ./org.apache.syncope/syncope-core/CVE-2018-17186.json
> ./org.apache.syncope/syncope-core/CVE-2020-1961.json
> ./org.apache.syncope/syncope-core/CVE-2018-17184.json
> ./org.apache.syncope/syncope-core/CVE-2018-1322.json
> ./org.apache.syncope/syncope-core/CVE-2018-1321.json
177a197,203
> ./org.apache.cxf.fediz/fediz-spring2/CVE-2017-12631.json
> ./org.apache.cxf.fediz/fediz-spring2/CVE-2018-8038.json
> ./org.apache.cxf.fediz/fediz-spring2/CVE-2016-4464.json
> ./org.apache.cxf.fediz/fediz-spring2/CVE-2015-5175.json
> ./org.apache.cxf.fediz/fediz-spring2/CVE-2017-7661.json
> ./org.apache.cxf.fediz/fediz-spring3/CVE-2017-12631.json
> ./org.apache.cxf.fediz/fediz-spring3/CVE-2018-8038.json
232a259,260
> ./org.opencrx/opencrx-core-config/CVE-2021-25959.json
> ./org.opencrx/opencrx-core-models/CVE-2021-25959.json
237a266
> ./org.jboss.resteasy/resteasy-client-microprofile/CVE-2020-25633.json
239a269
> ./org.jboss.resteasy/resteasy-jaxrs-all/CVE-2020-1695.json
260a291
> ./org.apache.cocoon/cocoon-core/CVE-2020-11991.json
261a293
> ./net.minidev/json-smart-mini/CVE-2021-27568.json
269a302,335
> ./io.netty/netty-codec/CVE-2020-11612.json
> ./io.netty/netty-codec/CVE-2021-37137.json
> ./io.netty/netty-codec/CVE-2021-21409.json
> ./io.netty/netty-codec/CVE-2021-21290.json
> ./io.netty/netty-codec/CVE-2021-37136.json
> ./io.netty/netty-all/CVE-2020-11612.json
> ./io.netty/netty-all/CVE-2020-7238.json
> ./io.netty/netty-all/CVE-2019-20445.json
> ./io.netty/netty-all/CVE-2019-16869.json
> ./io.netty/netty-all/CVE-2016-4970.json
> ./io.netty/netty-all/CVE-2021-21409.json
> ./io.netty/netty-all/CVE-2019-20444.json
> ./io.netty/netty-all/CVE-2021-21290.json
> ./io.netty/netty-codec-http2/CVE-2020-11612.json
> ./io.netty/netty-codec-http2/CVE-2020-7238.json
> ./io.netty/netty-codec-http2/CVE-2021-21409.json
> ./io.netty/netty-codec-http2/CVE-2021-21290.json
> ./io.netty/netty-codec-http2/CVE-2021-21295.json
> ./io.netty/netty-codec-http/CVE-2020-7238.json
> ./io.netty/netty-codec-http/CVE-2019-20445.json
> ./io.netty/netty-codec-http/CVE-2019-16869.json
> ./io.netty/netty-codec-http/CVE-2021-43797.json
> ./io.netty/netty-codec-http/CVE-2021-21409.json
> ./io.netty/netty-codec-http/CVE-2019-20444.json
> ./io.netty/netty-codec-http/CVE-2021-21290.json
> ./io.netty/netty-handler/CVE-2020-11612.json
> ./io.netty/netty-handler/CVE-2020-7238.json
> ./io.netty/netty-handler/CVE-2014-3488.json
> ./io.netty/netty-handler/CVE-2019-20445.json
> ./io.netty/netty-handler/CVE-2015-2156.json
> ./io.netty/netty-handler/CVE-2016-4970.json
> ./io.netty/netty-handler/CVE-2021-21409.json
> ./io.netty/netty-handler/CVE-2019-20444.json
> ./io.netty/netty-handler/CVE-2021-21290.json
298a365,366
> ./org.http4s/http4s-server_2.12/CVE-2020-5280.json
> ./org.http4s/blaze-core_2.12/CVE-2021-21293.json
302a371
> ./org.http4s/blaze-core_2.13/CVE-2021-21293.json
303a373
> ./org.http4s/blaze-core_2.11/CVE-2021-21293.json
316a387
> ./org.apache.shiro/shiro-spring-boot-starter/CVE-2020-17523.json
334a406,407
> ./org.rundeck/rundeck-core/CVE-2021-39133.json
> ./org.rundeck/rundeck-core/CVE-2021-39132.json
386a460,469
> ./com.vaadin/flow-client/CVE-2021-31406.json
> ./com.vaadin/flow-client/CVE-2020-36319.json
> ./com.vaadin/flow-client/CVE-2021-31405.json
> ./com.vaadin/flow-client/CVE-2021-31403.json
> ./com.vaadin/flow-client/CVE-2021-31408.json
> ./com.vaadin/flow-client/CVE-2021-31407.json
> ./com.vaadin/flow-client/CVE-2020-36321.json
> ./com.vaadin/flow-client/CVE-2020-36320.json
> ./com.vaadin/flow-client/GMS-2021-139.json
> ./com.vaadin/flow-client/CVE-2021-31404.json
390a474,485
> ./com.vaadin/flow-server/CVE-2021-31406.json
> ./com.vaadin/flow-server/GMS-2021-140.json
> ./com.vaadin/flow-server/CVE-2020-36319.json
> ./com.vaadin/flow-server/CVE-2018-25007.json
> ./com.vaadin/flow-server/CVE-2019-25027.json
> ./com.vaadin/flow-server/GMS-2021-142.json
> ./com.vaadin/flow-server/CVE-2021-31407.json
> ./com.vaadin/flow-server/CVE-2020-36321.json
> ./com.vaadin/flow-server/CVE-2021-31411.json
> ./com.vaadin/flow-server/CVE-2021-31404.json
> ./com.vaadin/flow-server/GMS-2021-141.json
> ./com.vaadin/flow-server/CVE-2021-31412.json
447a543,544
> ./org.apache.dubbo/dubbo-parent/CVE-2020-11995.json
> ./org.apache.dubbo/dubbo-common/CVE-2020-1948.json
455a553,554
> ./org.apache.pdfbox/pdfbox-parent/CVE-2021-31811.json
> ./org.apache.pdfbox/pdfbox-parent/CVE-2021-31812.json
464a564
> ./com.ctrip.framework.apollo/apollo-core/CVE-2020-15170.json
470a571
> ./org.openapitools/openapi-generator-online/CVE-2021-21428.json
474a576
> ./org.openapitools/openapi-generator-maven-plugin/CVE-2021-21429.json
479a582,583
> ./org.apache.sshd/sshd-mina/CVE-2021-30129.json
> ./org.apache.sshd/sshd-core/CVE-2021-30129.json
482a587
> ./org.apache.accumulo/accumulo-core/CVE-2020-17533.json
484a590,591
> ./org.springframework.security.oauth/spring-security-oauth-parent/CVE-2022-22969.json
> ./org.springframework.security.oauth/spring-security-oauth2/CVE-2016-4977.json
499a607,609
> ./org.apache.kylin/kylin-server-base/CVE-2020-13926.json
> ./org.apache.kylin/kylin-server-base/CVE-2020-13925.json
> ./org.apache.kylin/kylin-server-base/CVE-2020-1937.json
510a621
> ./org.apache.kylin/kylin-core-common/CVE-2020-1956.json
597a709,710
> ./org.apache.kafka/kafka_2.10/CVE-2021-26291.json
> ./org.apache.kafka/kafka_2.10/CVE-2017-12610.json
604a718
> ./org.wildfly.openssl/wildfly-openssl-java/CVE-2020-25644.json
635a750
> ./org.apache.tika/tika-parsers/CVE-2019-10093.json
637a753
> ./org.apache.tika/tika-core/CVE-2018-11762.json
638a755
> ./org.apache.tika/tika-core/CVE-2018-1339.json
640a758,765
> ./org.apache.tika/tika-core/CVE-2016-4434.json
> ./org.apache.tika/tika-core/CVE-2015-3271.json
> ./org.apache.tika/tika-core/CVE-2018-8017.json
> ./org.apache.tika/tika-core/CVE-2016-6809.json
> ./org.apache.tika/tika-core/CVE-2018-11796.json
> ./org.apache.tika/tika-core/CVE-2018-1335.json
> ./org.apache.tika/tika-core/CVE-2018-11761.json
> ./org.apache.tika/tika-core/CVE-2018-1338.json
669a795
> ./org.apache.camel/camel-jacksonxml/CVE-2016-9571.json
715a842
> ./org.bouncycastle/bcprov-jdk15on/CVE-2020-15522.json
716a844,845
> ./org.bouncycastle/bcprov-jdk15to18/CVE-2020-15522.json
> ./org.bouncycastle/bcprov-jdk15to18/CVE-2020-28052.json
758a888,889
> ./com.typesafe.play/play-java/CVE-2020-27196.json
> ./com.typesafe.play/play-java/CVE-2020-26883.json
795a927
> ./org.neo4j/neo4j-enterprise/CVE-2018-18389.json
822a955
> ./org.opencastproject/opencast-common-jpa-impl/CVE-2020-5229.json
1011a1145,1156
> ./org.apache.tomcat/tomcat-coyote/CVE-2013-2185.json
> ./org.apache.tomcat/tomcat-coyote/CVE-2021-24122.json
> ./org.apache.tomcat/tomcat-coyote/CVE-2020-17527.json
> ./org.apache.tomcat/tomcat-coyote/CVE-2014-0050.json
> ./org.apache.tomcat/tomcat-coyote/CVE-2020-13934.json
> ./org.apache.tomcat/tomcat-coyote/CVE-2020-1938.json
> ./org.apache.tomcat/tomcat-coyote/CVE-2020-11996.json
> ./org.apache.tomcat/tomcat-coyote/CVE-2020-1935.json
> ./org.apache.tomcat/tomcat-coyote/CVE-2021-30639.json
> ./org.apache.tomcat/tomcat-coyote/CVE-2020-13943.json
> ./org.apache.tomcat/tomcat-coyote/CVE-2021-33037.json
> ./org.apache.tomcat/tomcat-coyote/CVE-2020-13935.json
1050a1196,1204
> ./org.apache.tomcat/tomcat-util/CVE-2018-8037.json
> ./org.apache.tomcat/tomcat-util/CVE-2019-0232.json
> ./org.apache.tomcat/tomcat-util/CVE-2019-10072.json
> ./org.apache.tomcat/tomcat-util/CVE-2021-24122.json
> ./org.apache.tomcat/tomcat-util/CVE-2020-0822.json
> ./org.apache.tomcat/tomcat-util/CVE-2018-1336.json
> ./org.apache.tomcat/tomcat-util/CVE-2019-0199.json
> ./org.apache.tomcat/tomcat-util/CVE-2020-1935.json
> ./org.apache.tomcat/tomcat-util/CVE-2019-0221.json
1055a1210,1216
> ./org.apache.tomcat/tomcat-catalina/CVE-2021-24122.json
> ./org.apache.tomcat/tomcat-catalina/CVE-2020-17527.json
> ./org.apache.tomcat/tomcat-catalina/CVE-2017-7674.json
> ./org.apache.tomcat/tomcat-catalina/CVE-2017-12616.json
> ./org.apache.tomcat/tomcat-catalina/CVE-2016-3092.json
> ./org.apache.tomcat/tomcat-catalina/CVE-2017-12615.json
> ./org.apache.tomcat/tomcat-catalina/CVE-2020-9484.json
1113a1275
> ./org.apache.knox/gateway-service-knoxsso/CVE-2021-42357.json
1134a1297,1299
> ./org.apache.tomee/tomee-webapp/CVE-2020-13931.json
> ./org.apache.tomee/tomee-webapp/CVE-2021-40690.json
> ./org.apache.tomee/tomee-webapp/CVE-2021-33037.json
1205a1371
> ./org.apache.dolphinscheduler/dolphinscheduler-server/CVE-2021-27644.json
1219a1386
> ./org.apache.velocity/velocity-engine-parent/CVE-2020-13936.json
1222a1390,1391
> ./org.apache.jena/jena-fuseki/CVE-2021-33192.json
> ./org.apache.jena/jena-core/CVE-2021-39239.json
1261a1431,1435
> ./org.apache.poi/poi-ooxml/CVE-2022-26336.json
> ./org.apache.poi/poi-ooxml/CVE-2017-5644.json
> ./org.apache.poi/poi-ooxml/CVE-2019-12415.json
> ./org.apache.poi/poi-ooxml/CVE-2014-3574.json
> ./org.apache.poi/poi-ooxml/CVE-2014-3529.json
1267a1442
> ./org.apache.poi/poi-scratchpad/CVE-2022-26336.json
1270a1446
> ./org.apache.velocity.tools/velocity-tools-parent/CVE-2020-13959.json
1294a1471,1472
> ./org.xwiki.platform/xwiki-platform-web-templates/CVE-2022-23622.json
> ./org.xwiki.platform/xwiki-platform-web-templates/CVE-2022-24819.json
1307a1486
> ./org.eclipse.rdf4j/rdf4j-runtime/CVE-2018-1000644.json
1327a1507,1508
> ./org.xwiki.commons/xwiki-commons-core/CVE-2021-32621.json
> ./org.xwiki.commons/xwiki-commons-core/CVE-2021-32620.json
1330a1512,1515
> ./org.jeecgframework.boot/jeecg-boot-base-core/CVE-2022-22880.json
> ./org.jeecgframework.boot/jeecg-boot-base-core/CVE-2022-22881.json
> ./org.jeecgframework.boot/jeecg-boot-base-core/CVE-2021-46089.json
> ./org.jeecgframework.boot/jeecg-boot-base-core/CVE-2021-44585.json
1341a1527,1532
> ./org.apache.spark/spark-core_2.10/CVE-2018-1334.json
> ./org.apache.spark/spark-core_2.10/CVE-2018-11770.json
> ./org.apache.spark/spark-core_2.10/CVE-2017-12612.json
> ./org.apache.spark/spark-core_2.10/CVE-2018-8024.json
> ./org.apache.spark/spark-core_2.10/CVE-2018-17190.json
> ./org.apache.spark/spark-core_2.10/CVE-2017-7678.json
1355a1547,1553
> ./org.apache.spark/spark-core_2.11/CVE-2018-1334.json
> ./org.apache.spark/spark-core_2.11/CVE-2018-11770.json
> ./org.apache.spark/spark-core_2.11/CVE-2017-12612.json
> ./org.apache.spark/spark-core_2.11/CVE-2018-8024.json
> ./org.apache.spark/spark-core_2.11/CVE-2018-17190.json
> ./org.apache.spark/spark-core_2.11/CVE-2017-7678.json
> ./org.apache.spark/spark-core_2.11/CVE-2019-10099.json
1423a1622,1625
> ./org.apache.archiva/archiva-webapp/CVE-2020-9495.json
> ./org.apache.archiva/archiva-webapp/CVE-2019-0214.json
> ./org.apache.archiva/archiva-webapp/CVE-2017-5657.json
> ./org.apache.archiva/archiva-webapp/CVE-2019-0213.json
1526a1729,1731
> ./org.jenkins-ci.plugins/gitlab-oauth/CVE-2020-2228.json
> ./org.jenkins-ci.plugins/gitlab-oauth/CVE-2022-27206.json
> ./org.jenkins-ci.plugins/gitlab-oauth/CVE-2022-25196.json
1564a1770,1773
> ./org.jenkins-ci.plugins/fortify-on-demand-uploader/CVE-2019-10449.json
> ./org.jenkins-ci.plugins/fortify-on-demand-uploader/CVE-2020-2204.json
> ./org.jenkins-ci.plugins/fortify-on-demand-uploader/CVE-2020-2203.json
> ./org.jenkins-ci.plugins/fortify-on-demand-uploader/CVE-2020-2202.json
1571a1781
> ./org.jenkins-ci.plugins/deployer-framework/CVE-2020-2227.json
1728a1939
> ./org.jenkins-ci.plugins/fortify-cloudscan-jenkins-plugin/CVE-2018-1000607.json
1767a1979,1981
> ./org.jenkins-ci.plugins/github-branch-source/CVE-2018-1000185.json
> ./org.jenkins-ci.plugins/github-branch-source/CVE-2017-1000087.json
> ./org.jenkins-ci.plugins/github-branch-source/CVE-2017-1000091.json
1815a2030
> ./org.jenkins-ci.plugins/git-parameter/CVE-2022-29040.json
1855a2071
> ./org.codehaus.castor/castor-xml/CVE-2014-3004.json
1856a2073
> ./org.apache.synapse/synapse-core/CVE-2017-15708.json
1872a2090,2096
> ./org.apache.cxf/cxf-rt-rs-security-xml/CVE-2020-13954.json
> ./org.apache.cxf/cxf-rt-rs-security-xml/CVE-2017-5653.json
> ./org.apache.cxf/cxf-rt-rs-security-xml/CVE-2021-22696.json
> ./org.apache.cxf/cxf-rt-rs-security-xml/CVE-2014-3584.json
> ./org.apache.cxf/cxf-rt-ws-security/CVE-2012-5633.json
> ./org.apache.cxf/cxf-rt-ws-security/CVE-2017-5656.json
> ./org.apache.cxf/cxf-rt-ws-security/CVE-2013-0239.json
1893a2118,2142
> ./org.apache.cxf/cxf-rt-bindings-soap/CVE-2012-3451.json
> ./org.apache.cxf/cxf-rt-transports-http/CVE-2020-13954.json
> ./org.apache.cxf/cxf-rt-transports-http/CVE-2021-22696.json
> ./org.apache.cxf/cxf-rt-transports-http/CVE-2016-6812.json
> ./org.apache.cxf/cxf-parent/CVE-2013-2160.json
> ./org.apache.cxf/cxf-rt-rs-json-basic/CVE-2021-30468.json
> ./org.apache.cxf/cxf-core/CVE-2020-13954.json
> ./org.apache.cxf/cxf-core/CVE-2020-1954.json
> ./org.apache.cxf/cxf-core/CVE-2021-22696.json
> ./org.apache.cxf/cxf-core/CVE-2019-12406.json
> ./org.apache.cxf/cxf-core/CVE-2019-12419.json
> ./org.apache.cxf/cxf-core/CVE-2019-12423.json
> ./org.apache.cxf/cxf-core/CVE-2018-8039.json
> ./org.apache.cxf/cxf-core/CVE-2019-17573.json
> ./org.apache.cxf/cxf-rt-rs-security-oauth2/CVE-2021-22696.json
> ./org.apache.cxf/cxf-rt-security/CVE-2014-3623.json
> ./org.apache.cxf/cxf-rt-rs-security-jose/CVE-2017-3156.json
> ./org.apache.cxf/cxf-api/CVE-2020-13954.json
> ./org.apache.cxf/cxf-api/CVE-2021-22696.json
> ./org.apache.cxf/cxf-api/CVE-2012-3451.json
> ./org.apache.cxf/cxf-api/CVE-2013-2160.json
> ./org.apache.cxf/cxf-rt-frontend-jaxrs/CVE-2020-13954.json
> ./org.apache.cxf/cxf-rt-frontend-jaxrs/CVE-2021-22696.json
> ./org.apache.cxf/cxf-rt-frontend-jaxrs/CVE-2017-12624.json
> ./org.apache.cxf/cxf-rt-management/CVE-2020-1954.json
1931a2181
> ./org.esigate/esigate-core/CVE-2018-1000854.json
2059a2310,2311
> ./org.springframework.boot/spring-boot-starter-web/GMS-2022-560.json
> ./org.springframework.boot/spring-boot-starter-webflux/GMS-2022-561.json
2062a2315,2316
> ./io.projectreactor.netty/reactor-netty-http/CVE-2020-5403.json
> ./io.projectreactor.netty/reactor-netty-http/CVE-2020-5404.json
2067a2322
> ./com.atlassian.connect/atlassian-connect-spring-boot-starter/CVE-2021-26074.json
2113a2369
> ./org.apache.pulsar/pulsar-broker-common/CVE-2021-41571.json
2126a2383
> ./io.jooby/jooby-netty/CVE-2020-7622.json
2166a2424
> ./org.exist-db/exist-core/CVE-2018-1000823.json
2174a2433
> ./org.apache.juddi/juddi-client/CVE-2015-5241.json
2175a2435
> ./org.apache.juddi/juddi-core/CVE-2021-37578.json
2196a2457,2465
> ./org.apache.storm/storm-kafka/CVE-2018-11779.json
> ./org.apache.storm/storm-kafka-client/CVE-2018-11779.json
> ./org.apache.storm/storm-server/CVE-2021-40865.json
> ./org.apache.storm/storm-server/CVE-2021-38294.json
> ./org.apache.storm/storm-core/CVE-2018-8008.json
> ./org.apache.storm/storm-core/CVE-2019-0202.json
> ./org.apache.storm/storm-core/CVE-2018-1331.json
> ./org.apache.storm/storm-core/CVE-2017-9799.json
> ./org.apache.storm/storm-core/CVE-2018-1332.json
2221a2491,2492
> ./org.eclipse.hawkbit/hawkbit-boot-starter-dmf-api/CVE-2019-10240.json
> ./org.eclipse.hawkbit/hawkbit-boot-starter-ddi-api/CVE-2019-10240.json
2223a2495
> ./org.eclipse.hawkbit/hawkbit-boot-starter-mgmt-api/CVE-2019-10240.json
2227a2500
> ./org.eclipse.hawkbit/hawkbit-boot-starter-mgmt-ui/CVE-2019-10240.json
2406a2680,2699
> ./org.apache.hive/hive-exec/CVE-2016-3083.json
> ./org.apache.hive/hive-exec/CVE-2014-0228.json
> ./org.apache.hive/hive-exec/CVE-2018-11777.json
> ./org.apache.hive/hive-exec/CVE-2018-1284.json
> ./org.apache.hive/hive-exec/CVE-2015-7521.json
> ./org.apache.hive/hive-exec/CVE-2017-12625.json
> ./org.apache.hive/hive-exec/CVE-2015-1772.json
> ./org.apache.hive/hive-exec/CVE-2018-1315.json
> ./org.apache.hive/hive-service/CVE-2016-3083.json
> ./org.apache.hive/hive-service/CVE-2014-0228.json
> ./org.apache.hive/hive-service/CVE-2020-1926.json
> ./org.apache.hive/hive-service/CVE-2018-1284.json
> ./org.apache.hive/hive-service/CVE-2015-7521.json
> ./org.apache.hive/hive-service/CVE-2017-12625.json
> ./org.apache.hive/hive-service/CVE-2015-1772.json
> ./org.apache.hive/hive-service/CVE-2018-1315.json
> ./org.apache.hive/hive-jdbc/CVE-2016-3083.json
> ./org.apache.hive/hive-jdbc/CVE-2017-12625.json
> ./org.apache.hive/hive-jdbc/CVE-2018-1282.json
> ./org.apache.hive/hive-jdbc/CVE-2018-1314.json
2562a2856,2857
> ./org.apache.nifi/nifi-security/CVE-2019-10080.json
> ./org.apache.nifi/nifi-web-api/CVE-2019-10083.json
2591a2887
> ./org.apache.nifi/nifi-framework-cluster-protocol/CVE-2017-15703.json
2597a2894
> ./org.springframework.cloud/spring-cloud-netflix-zuul/CVE-2021-22113.json
2603a2901
> ./org.springframework.cloud/spring-cloud-netflix-hystrix-dashboard/CVE-2021-22053.json
2654a2953,3023
> ./org.springframework/spring-core/CVE-2016-9878.json
> ./org.springframework/spring-core/CVE-2016-1000027.json
> ./org.springframework/spring-core/CVE-2018-1199.json
> ./org.springframework/spring-core/CVE-2022-22968.json
> ./org.springframework/spring-core/CVE-2019-11272.json
> ./org.springframework/spring-core/CVE-2011-2730.json
> ./org.springframework/spring-core/CVE-2018-1257.json
> ./org.springframework/spring-core/CVE-2018-1270.json
> ./org.springframework/spring-core/CVE-2019-3795.json
> ./org.springframework/spring-core/CVE-2015-0201.json
> ./org.springframework/spring-core/CVE-2009-1190.json
> ./org.springframework/spring-core/CVE-2021-22060.json
> ./org.springframework/spring-core/CVE-2018-11040.json
> ./org.springframework/spring-core/CVE-2018-15756.json
> ./org.springframework/spring-core/CVE-2018-1271.json
> ./org.springframework/spring-core/CVE-2010-1622.json
> ./org.springframework/spring-core/CVE-2015-3192.json
> ./org.springframework/spring-core/CVE-2016-5007.json
> ./org.springframework/spring-core/CVE-2018-11039.json
> ./org.springframework/spring-core/CVE-2020-5421.json
> ./org.springframework/spring-core/CVE-2018-1275.json
> ./org.springframework/spring-core/GMS-2022-559.json
> ./org.springframework/spring-core/CVE-2018-1258.json
> ./org.springframework/spring-core/CVE-2022-22950.json
> ./org.springframework/spring-core/CVE-2018-1272.json
> ./org.springframework/spring-core/CVE-2015-5211.json
> ./org.springframework/spring-framework-bom/CVE-2020-5421.json
> ./org.springframework/spring-web/CVE-2013-7315.json
> ./org.springframework/spring-web/CVE-2014-0225.json
> ./org.springframework/spring-web/CVE-2013-6429.json
> ./org.springframework/spring-web/CVE-2018-11040.json
> ./org.springframework/spring-web/CVE-2018-15756.json
> ./org.springframework/spring-web/CVE-2015-3192.json
> ./org.springframework/spring-web/CVE-2020-5421.json
> ./org.springframework/spring-web/CVE-2013-6430.json
> ./org.springframework/spring-web/CVE-2020-5398.json
> ./org.springframework/spring-web/CVE-2015-5211.json
> ./org.springframework/spring-beans/GMS-2022-558.json
> ./org.springframework/spring-webflux/CVE-2018-15756.json
> ./org.springframework/spring-webflux/CVE-2022-22965.json
> ./org.springframework/spring-webflux/CVE-2020-5421.json
> ./org.springframework/spring-webflux/CVE-2021-22118.json
> ./org.springframework/spring-webflux/CVE-2020-5397.json
> ./org.springframework/spring-webflux/CVE-2020-5398.json
> ./org.springframework/spring-webflux/CVE-2018-1272.json
> ./org.springframework/spring-oxm/CVE-2014-0225.json
> ./org.springframework/spring-oxm/CVE-2020-5421.json
> ./org.springframework/spring-oxm/CVE-2013-4152.json
> ./org.springframework/spring-oxm/CVE-2013-6430.json
> ./org.springframework/spring-oxm/CVE-2020-5397.json
> ./org.springframework/spring-webmvc/CVE-2016-9878.json
> ./org.springframework/spring-webmvc/CVE-2014-1904.json
> ./org.springframework/spring-webmvc/CVE-2014-3625.json
> ./org.springframework/spring-webmvc/CVE-2018-15801.json
> ./org.springframework/spring-webmvc/CVE-2018-15756.json
> ./org.springframework/spring-webmvc/CVE-2018-1271.json
> ./org.springframework/spring-webmvc/CVE-2018-11039.json
> ./org.springframework/spring-webmvc/CVE-2022-22965.json
> ./org.springframework/spring-webmvc/CVE-2020-5421.json
> ./org.springframework/spring-webmvc/CVE-2013-6430.json
> ./org.springframework/spring-webmvc/CVE-2020-5397.json
> ./org.springframework/spring-webmvc/CVE-2020-5398.json
> ./org.springframework/spring-webmvc/CVE-2014-0054.json
> ./org.springframework/spring-webmvc/CVE-2022-22950.json
> ./org.springframework/spring-webmvc/CVE-2018-1272.json
> ./org.springframework/spring-webmvc/CVE-2014-3578.json
> ./org.springframework/spring-messaging/CVE-2018-1257.json
> ./org.springframework/spring-messaging/CVE-2018-1270.json
> ./org.springframework/spring-messaging/CVE-2020-5421.json
> ./org.springframework/spring-websocket/CVE-2015-0201.json
> ./org.springframework/spring-websocket/CVE-2015-5211.json
2663a3033
> ./ca.uhn.hapi.fhir/hapi-fhir-base/CVE-2019-12741.json
2666a3037,3038
> ./ca.uhn.hapi.fhir/hapi-fhir-jpaserver-base/CVE-2021-32053.json
> ./ca.uhn.hapi.fhir/hapi-fhir-testpage-overlay/CVE-2020-24301.json
2733a3106,3116
> ./org.apache.ozone/ozone-main/CVE-2021-39234.json
> ./org.apache.ozone/ozone-main/CVE-2021-39233.json
> ./org.apache.ozone/ozone-main/CVE-2021-39231.json
> ./org.apache.ozone/ozone-main/CVE-2021-39232.json
> ./org.apache.ozone/ozone-main/CVE-2021-39236.json
> ./org.apache.ozone/ozone-main/CVE-2021-41532.json
> ./org.apache.ozone/ozone-main/CVE-2021-39235.json
> ./org.apache.ozone/ozone-main/CVE-2021-36372.json
> ./org.apache.ozone/ozone-datanode/CVE-2021-39233.json
> ./org.apache.ozone/ozone-datanode/CVE-2021-39231.json
> ./org.apache.ozone/ozone-datanode/CVE-2021-39235.json
2742a3126
> ./org.apache.wicket/wicket-util/CVE-2021-23937.json
2743a3128
> ./org.apache.wicket/wicket-core/CVE-2021-23937.json
2751a3137,3140
> ./com.typesafe.akka/akka-http-core_2.12/CVE-2017-1000118.json
> ./com.typesafe.akka/akka-http-core_2.12/CVE-2018-16131.json
> ./com.typesafe.akka/akka-http-core_2.11/CVE-2017-1000118.json
> ./com.typesafe.akka/akka-http-core_2.11/CVE-2018-16131.json
2753a3143,3146
> ./com.typesafe.akka/akka-actor_2.11/CVE-2018-16115.json
> ./com.typesafe.akka/akka-actor_2.11/CVE-2017-1000034.json
> ./com.typesafe.akka/akka-actor_2.12/CVE-2018-16115.json
> ./com.typesafe.akka/akka-actor_2.12/CVE-2017-1000034.json
2774a3168,3169
> ./org.apache/camel-core/CVE-2020-11971.json
> ./org.apache/camel-management/CVE-2020-11971.json
2787a3183
> ./com.twitter/twitter-server_2.12/CVE-2020-35774.json
2794a3191,3194
> ./org.apache.shenyu/shenyu-common/CVE-2022-23944.json
> ./org.apache.shenyu/shenyu-common/CVE-2022-23945.json
> ./org.apache.shenyu/shenyu-common/CVE-2021-45029.json
> ./org.apache.shenyu/shenyu-common/CVE-2022-23223.json
2819a3220,3221
> ./org.jboss.weld/weld-core-impl/CVE-2014-8122.json
> ./org.jboss.weld/weld-core-bom/CVE-2014-8122.json
2880a3283
> ./org.jboss.seam/jboss-seam-remoting/CVE-2014-0149.json
2897a3301,3302
> ./org.apache.qpid/qpid-broker-plugins-amqp-0-8-protocol/CVE-2016-4432.json
> ./org.apache.qpid/qpid-broker-plugins-amqp-1-0-protocol/CVE-2016-4432.json
```

</details>

## Related issues:
- https://github.com/aquasecurity/trivy/issues/2089

### CODE REVIEW

File: glad/glad.go [hunk 1]
```diff
@@ -136,12 +139,19 @@ func (u Updater) walkDir(root string) error {
...
+			// => go/github.com/apache/thrift/lib/go/thrift
+			// go/github.com/apache/thrift/lib/go/thrift and go/github.com/apache/thrift-mini
+			// => skip this advisory
+			suffix := strings.TrimPrefix(adv.PackageSlug, strings.TrimSuffix(a.PackageSlug, "/"))
+			if suffix == "/" { // same slugs
+				continue
+			}
+			if strings.HasPrefix(suffix, "/") {
+				return a.PackageSlug
+			}
```
Comment (knqyf263) [inline glad/glad.go:154]: What if just adding `/` at last?

```
slug := adv.PackageSlug + "/"
```

Or

```
if !strings.HasSuffix(adv.PackageSlug, "/") {
    slug := adv.PackageSlug + "/"
}
```


Then, `go/github.com/apache/thrift/lib/go/thrift` and `go/github.com/apache/thrift-mini` will not match.

File: glad/glad.go [hunk 2]
```diff
@@ -112,20 +112,23 @@ func (u Updater) walkDir(root string) error {
...
 		adv.Identifier = strings.ToUpper(adv.Identifier)
 
-		slug := u.searchPrefix(adv, advisories)
-		if slug != "" {
-			// Update the package_slug to flatten nested packages
-			// e.g.  go/k8s.io/kubernetes => go/k8s.io/kubernetes
-			//       go/k8s.io/kubernetes/pkg/kubelet/kuberuntime => go/k8s.io/kubernetes
-			adv.PackageSlug = slug
+		// Only 'go' package slugs need to be updated
+		if strings.HasPrefix(adv.PackageSlug, "go") {
```
Comment (knqyf263) [inline glad/glad.go:121]:
```suggestion
		if strings.HasPrefix(adv.PackageSlug, "go/") {
```

Comment (DmitriyLewen) [reply glad/glad.go]: Done

File: glad/glad.go (hunk 1 above)
Comment (DmitriyLewen) [reply glad/glad.go]: I used regular expression. I think it is easier to understand. Can you take a look?

Comment (knqyf263) [reply glad/glad.go]: Looks like `HasPrefix` is better. Any reason regexp is preferred?

Comment (DmitriyLewen) [reply glad/glad.go]: I thought regexp would be easier to understand, but i updated PR without regexp and followed your recommendations. Looks better. Thanks a lot!

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**